Sources/SourceControl/GitRepository.swift

```diff
@@ -105,20 +105,37 @@ public struct GitRepositoryProvider: RepositoryProvider, Cancellable {
         progress: FetchProgress.Handler? = nil
     ) throws -> String {
         if let progress {
-            var stdoutBytes: [UInt8] = [], stderrBytes: [UInt8] = []
+            nonisolated(unsafe) var stdoutBytes: [UInt8] = []
+            nonisolated(unsafe) var stderrBytes: [UInt8] = []
```

Comment:
I'm concerned we're trading concurrency warnings for `nonisolated(unsafe)` access on mutable arrays. Have you tried using new `AsyncProcess` APIs based on `AsyncSequence` instead? We'd have to re-enable corresponding tests on CI first: https://github.com/swiftlang/swift-package-manager/blob/d70b78bb2332855163c25ef769d2310aceaf4c17/Tests/BasicsTests/AsyncProcessTests.swift#L398

Reply:
why this test is disabled?

Reply:
There seem to be non-deterministic failures in some CI jobs for it, so those have to be fixed first.